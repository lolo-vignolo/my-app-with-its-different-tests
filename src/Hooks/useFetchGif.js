import { useState, useEffect } from "react";
import getGifs from "../helpers/getGifs";



export const useFetchGif = (Category)=>{

    const [state, setstate] = useState({
        data : [],
        loading: true
    });

    useEffect(() => {

        getGifs(Category)
        .then(imgs =>{
            
                setstate({
                    data: imgs,
                    loading : false
                });
                 
           

        })}
            
       
     
    , [Category]);

    return state;
} 

