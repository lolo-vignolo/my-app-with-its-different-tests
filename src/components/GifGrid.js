
import React  from 'react'
import { useFetchGif } from '../Hooks/useFetchGif'
import PropTypes from 'prop-types';

import GifGridItem from './GifGridItem'




const GifGrid = ({Category})=> {

    // const [images , setImages] = useState([])

 

    const {data:images, loading} = useFetchGif(Category);


    
    return (
         <>
           <h3 calssName="animate__animated animate__fadeInLeft">{Category}</h3>

           {loading && <p className="animate__animated animate__flash">loading...</p>}

              <div className = "card-grid">
                {
                    images.map(image =>
                        (
                          <GifGridItem
                            key= {image.id}
                            {...image} 

                            />

                        )
                    )
                }
            
            </div>
        </>
    )
}

GifGrid.propType ={
    Category : PropTypes.string.isRequired,
}

export default GifGrid
