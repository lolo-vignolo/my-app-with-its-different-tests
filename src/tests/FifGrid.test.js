import { shallow } from 'enzyme';
import React from 'react';
import GifGrid from '../components/GifGrid';
import { useFetchGif } from '../Hooks/useFetchGif';
jest.mock('../Hooks/useFetchGif');



describe("testing a component with properties, and a fetch inside", ()=>{

    const category = "Peaky blinders"

    test('observar si se renderiza correctamente ', () => {

     
        const wrapper = shallow( <GifGrid Category = {category} />)
        expect(wrapper).toMatchSnapshot();


       })

       


        test('it should shows items when images are charged with useFeTCHgIF ', () => {
            
            const gifs = [{
                id:"abc",
                url:"http://something.jnp",
                title : "testing my code"
            }]

            useFetchGif.mockReturnValue({
                data: gifs,
                loading: true
               });   
    
            
        })
        


        
    })



    


