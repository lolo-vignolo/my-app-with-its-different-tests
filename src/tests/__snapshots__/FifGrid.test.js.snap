// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing a component with properties, and a fetch inside it should shows items when images are charged with useFeTCHgIF  1`] = `
<Fragment>
  <h3
    calssName="animate__animated animate__fadeInLeft"
  >
    Peaky blinders
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading...
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="testing my code"
      url="http://something.jnp"
    />
  </div>
</Fragment>
`;

exports[`testing a component with properties, and a fetch inside observar si se renderiza correctamente  1`] = `
<Fragment>
  <h3
    calssName="animate__animated animate__fadeInLeft"
  >
    Peaky blinders
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
